ffcx/ir/integral.py

```diff
@@ -112,10 +113,22 @@ def compute_integral_ir(cell, integral_type, entitytype, integrands, argument_sh
                 if deps:
                     v['expression'] = v['expression']._ufl_expr_reconstruct_(*deps)
 
-            # Rebuild scalar target expressions and graph (this may be
-            # overkill and possible to optimize away if it turns out to be
-            # costly)
-            expression = S.nodes[S_targets[0]]['expression']
+            if value_rank == 0:
+                # Rebuild scalar target expressions and graph (this may be
+                # overkill and possible to optimize away if it turns out to be
+                # costly)
+                expression = S.nodes[S_targets[0]]['expression']
+            elif value_rank == 1 or value_rank == 2:
+                expressions = [None, ] * num_components
+                for target in S_targets:
+                    for comp in S.nodes[target]["component"]:
+                        expressions[comp] = S.nodes[target]["expression"]
```

Comment:
Can you add assert here, that this is not overwriting some component added earlier, `assert(expressions[comp] == None)` before. This should never be the case, just to be safe.